### PR TITLE
refactor metadata to dataclass

### PR DESCRIPTION
**Pull request by yanksyoon**
This PR is a refactor PR in preparation for https://github.com/canonical/upload-charm-docs/issues/9, Migration for existing docs.

Changes:
- change metadata dict to metadata dataclass
- add metadata required fields (name & docs key)
- add metadata required fields testing

### CODE REVIEW

**Comment by yanksyoon** [comment]
Fixing the Self-tests!

File: .gitignore [hunk 1]
```diff
@@ -127,3 +128,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+# VSCode settings
+.vscode
```
Comment (jdkandersson) [inline .gitignore:133]: Final new line?

Comment (yanksyoon) [reply .gitignore:133]: I've definitely gotta add that to my IDE formatter like black I keep missing those! 
My bad!

**Comment by yanksyoon** [comment]
Closing this PR, unable to read secrets from a forked repository pull request, my bad!

**Pull request closed without merging**